### DISCUSSION TITLE
How to Build and Install the VSCode Extension

### DISCUSSION THREAD

**Discussion by ballcoach12** [eclipse-langium/langium#1148] [Q&A] [answered]
Hey guys,

I have a grammar that is giving me problems and it might help me to have a railroad diagram generated from my grammar.  I see that this feature is in the upcoming `2.0.0` release, which is not out yet, so I want to know if I can feasibly build a VSIX for the VSCode package so that I can install it and get access to the new feature as a preview.

I don't see any npm package scripts that appear to output the VSIX.  If I recall correctly, VSIX is just an archive format, so maybe I need a different tool in order to generate it? How can I build the extension so that I can use it to preview the railroad diagram feature?

Thanks!

**Comment by msujew** [comment, marked as answer]
Hey @ballcoach12,

you can run `vsce package` (you will need to run `npm i -g vsce` first) in the `langium-vscode` directory to generate the `.vsix` file. Alternatively just run the `Run Grammar Extension` debug config and open your project in the newly opened vscode window. That should yield you the same effect.

**Reply by ballcoach12** [reply]
Thank you, Mark. Got it up and running quickly.  The railroad diagram is a nice addition!
